Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -2242,4 +2243,254 @@ - (void)testAddPropertyToReadOnlyRealmWithAsyncOpen {
     }
 }
 
+#pragma mark - Watch
+
+- (void)testWatch {
+    [self performWatchTest:nil];
+}
+
+- (void)testWatchAsync {
+    auto asyncQueue = dispatch_queue_create("io.realm.watchQueue", DISPATCH_QUEUE_CONCURRENT);
+    [self performWatchTest:asyncQueue];
+}
+
+- (void)performWatchTest:(nullable dispatch_queue_t)delegateQueue {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"watch collection and receive change event 5 times"];
+
+    RLMMongoClient *client = [self.anonymousUser mongoClientWithServiceName:@"mongodb1"];
+    RLMMongoDatabase *database = [client databaseWithName:@"test_data"];
+    __block RLMMongoCollection *collection = [database collectionWithName:@"Dog"];
+
+    RLMWatchTestUtility *testUtility = [[RLMWatchTestUtility alloc] initWithChangeEventCount:5
+                                                                                  completion:^(NSError * error) {
+        XCTAssertNil(error);
+        [expectation fulfill];
+    }];
+    __block RLMChangeStream *changeStream = [collection watchWithDelegate:testUtility delegateQueue:delegateQueue];
+
+    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        for (int i = 0; i < 5; i++) {
+            [collection insertOneDocument:@{@"name": @"fido"} completion:^(RLMObjectId * objectId, NSError * error) {
+                XCTAssertNil(error);
+                XCTAssertNotNil(objectId);
+            }];
+            [NSThread sleepForTimeInterval:2];
+            if (i == 4) {
+                [NSThread sleepForTimeInterval:10];
```

Comment:
Why does this need to sleep? Can't it just close the changestream on the main thread after waiting for expectations?